src/fwupd_receive_updates.py

```diff
@@ -0,0 +1,281 @@
+        regex_pattern -- pattern of the expected files
+        """
+        # Verify received files
+        for untrusted_f in os.listdir(files_path):
+            if not regex_pattern.match(untrusted_f):
+                raise TypeError(
+                    'Domain ' + self.source + ' sent unexpected file'
+                )
+            f = untrusted_f
+            assert '/' not in f
+            assert '\0' not in f
+            assert '\x1b' not in f
+            path_f = path.join(files_path, f)
+            if os.path.islink(path_f) or not os.path.isfile(path_f):
+                raise TypeError(
+                    'Domain ' + self.source + ' sent not regular file'
+                )
+
+    def _create_dirs(self, *args):
+        """Method creates directories.
+
+        Keyword arguments:
+        *args -- paths to be created
+        """
+        # Create directories for update files
+        qubes_gid = grp.getgrnam('qubes').gr_gid
+        self.old_umask = os.umask(0o002)
+        if args is None:
+            raise ValueError("Creating directories failed, no paths given.")
+        for file_path in args:
+            if not path.exists(file_path):
+                os.mkdir(file_path)
+                os.chown(file_path, -1, qubes_gid)
+                os.chmod(file_path, 0o0775)
+
+    def _extract_archive(self, archive_path, output_path):
+        """Extracts archive file to specified directory.
+
+        Keyword arguments:
+        archive_path -- absolute path to archive file
+        output_path -- absolute output directory
+        """
+        cmd_extract = [
+            "cabextract",
+            "-d",
+            "%s" % output_path,
+            "%s" % archive_path
+        ]
+        shutil.copy(archive_path, FWUPD_DOM0_UPDATES_DIR)
+        p = subprocess.Popen(cmd_extract, stdout=subprocess.PIPE)
+        p.communicate()[0].decode('ascii')
+        if p.returncode != 0:
+            raise Exception(
+                'cabextract: Error while extracting %s.' %
+                archive_path
+            )
+
+    def _gpg_verification(self, file_path):
+        """Inspects GPG signature.
+
+        Keyword argument:
+        file_path -- absolute path to inspected file
+        """
+        cmd_gpg = [
+            "gpg",
+            "--verify",
+            "%s.asc" % file_path,
+            "%s" % file_path,
+        ]
+        p = subprocess.Popen(
+            cmd_gpg,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        __, stderr = p.communicate()
+        verification = stderr.decode('ascii')
+        print(verification)
+        if p.returncode != 0:
+            raise Exception('gpg: Verification failed')
+        if not GPG_LVFS_REGEX.search(verification.strip()):
+            raise Exception(
+                'Domain updateVM sent not signed firmware: ' + file_path
+            )
+
+    def handle_fw_update(self, updatevm, shasum, filename):
```

Comment:
In some functions it is `shasum` in others `sha`. Let's be consistent if this represents the same thing.

Reply:
Fixed.